### DISCUSSION TITLE
Can we process Duckdb database file present on Blob Storage

### DISCUSSION THREAD

**Discussion by akash-goel** [duckdb/duckdb#7692] [General]
Hi Team,

We want to save duckdb database file on Azure blob storage and perform operation directly on blob .

`con = duckdb.connect("test.duckdb")`

So we would like to store "test.duckdb" directly on blob storage and run the DML queries on that.

Can you please guide , is it possible to create duckdb database file on blob and perform operation directly on that file ?

Regards,
Akash Goel

**Comment by Alex-Monahan** [comment]
Today, blob storage is not supported unfortunately! There are some challenges there: each file is re-written in its entirety on any change, so a single file database would be completely rewritten on every insert or update...

However, reading and writing to blob storage are supported with Parquet files, include Hive-style directories of parquet files. That approach is better suited to object storage!

**Reply by Neevan960** [reply]
can you share the documentation on how to attach the database residing in azure datalake
